### DISCUSSION TITLE
WiringPi - Raspberry Pi 4 % cpuinfo 'Hardware'

### DISCUSSION THREAD

**Discussion by Yeshu-777** [WiringPi/WiringPi#278] [Q&A] [answered]
Hi,

The /proc/cupinfo on "Bookworm" doesn't have the "Hardware" label, which WiringPi relies on currently.

A work around is to utilise a copy of 'cpuinfo' with the label added and change the build to point to this file instead and recompile.

Hardware	: BCM2835
Revision	: b03115
Serial	: 100000006be1a682
Model	: Raspberry Pi 4 Model B Rev 1.5

Has this been resolved in the current hosted vrsion?

Yeshu777

**Comment by mstroh76** [comment]
Hi,
where did you find that problem, which version do you use? This problem was fixed a long time ago.

**Comment by Yeshu-777** [comment]
Hi, 

Thanks for the reply..

Please see [https://forum.thegamecreators.com/thread/229480](url)

There is a post which reads...

Hey Yeshu, 

I found the trouble with WiringPi on Bookworm 32 and 64.

What happens is that the file /proc/cpuinfo in Bookworm is lacking a line that defines "hardware". I found this comparing the same file from a Bullseye install.
My solution is a lazy one, basically am tricking Wiring Pi into reading the bullseye cpuinfo file.
I only tested gpio digital outputs but it is highly likely all other functions will work. Maybe you can test with serial ( I2c or SPI if possible) and report back here.

Here is the procedure:

- Clone WiringPi git as in the previous post.
- Get a copy of bullseye /proc/cpuinfo file, or copy the attached file. It is the same file for Bullseye 32 and 64. Put the file in the pi wherever you feel convenient. I placed it on my home folder.
- make a folder to put the file in a "standard" location, and copy the file there, like so:

sudo mkdir /usr/share/legacywiringpi
sudo cp ~/cpuinfo /usr/share/legacywiringpi/cpuinfo
sudo chmod 444 /usr/share/legacywiringpi/cpuinfo

- Using your preferred editor, open the files "WiringPi/build" and "WiringPi/WiringPi/wiringPi.c" from the cloned repository.
- In both files change any reference to "/proc/cpuinfo" to "/usr/share/legacywiringpi/cpuinfo". On both files there are file open references and echo messages using that string.
- Save both files.
- now you can navigate to your cloned repository folder in a terminal and
./build
It should now be working and installed.

**Comment by mstroh76** [comment, marked as answer]
very old post _20th Jan 2024 19:10_
 This has been solved a long time ago ...
`#ifdef	DONT_CARE_ANYMORE`  😄 

code is still there for nostalgia and to keep the comments from Gordon:
https://github.com/WiringPi/WiringPi/blob/master/wiringPi/wiringPiLegacy.c

**Comment by Yeshu-777** [comment]
Hey,

Thanks for confirming.. great work guys..

Yeshu
